dali/test/python/test_operator_multipaste.py

```diff
@@ -0,0 +1,196 @@
+            in_idx_l[out_idx] = np.append(in_idx_l[out_idx], [in_idx], axis=0)
+            in_anchors_l[out_idx] = np.append(in_anchors_l[out_idx], [in_anchor], axis=0)
+            shapes_l[out_idx] = np.append(shapes_l[out_idx], [shape], axis=0)
+            out_anchors_l[out_idx] = np.append(out_anchors_l[out_idx], [out_anchor], axis=0)
+    return in_idx_l, in_anchors_l, shapes_l, out_anchors_l
+
+
+def numpyize(source_list):
+    print(source_list)
+    return lambda: source_list
+
+
+def get_pipeline(
+        batch_size=4,
+        in_size=None,
+        out_size=None,
+        even_paste_count=False,
+        k=4,
+        d_type=types.DALIDataType.UINT8,
+        no_intersections=True,
+):
+    pipe = Pipeline(batch_size=batch_size, num_threads=4, device_id=types.CPU_ONLY_DEVICE_ID)
+    with pipe:
+        input, _ = fn.file_reader(file_root=img_dir)
+        decoded = fn.image_decoder(input, device='cpu', output_type=types.RGB)
+        resized = fn.resize(decoded, resize_x=in_size[1], resize_y=in_size[0])
+        in_idx_l, in_anchors_l, shapes_l, out_anchors_l = prepare_cuts(
+            k, batch_size, in_size, out_size, even_paste_count, no_intersections)
+        in_idx = fn.external_source(numpyize(in_idx_l))
+        in_anchors = fn.external_source(numpyize(in_anchors_l))
+        shapes = fn.external_source(numpyize(shapes_l))
+        out_anchors = fn.external_source(numpyize(out_anchors_l))
+        kwargs = {
+            "in_ids": in_idx,
+            "in_anchors": in_anchors,
+            "out_anchors": out_anchors,
+            "shapes": shapes,
+            "output_size": out_size,
+            "dtype": d_type,
+            "no_intersections": no_intersections
+        }
+
+        pasted = fn.multi_paste(resized, **kwargs)
+        pipe.set_outputs(pasted, resized)
+    return pipe, in_idx_l, in_anchors_l, shapes_l, out_anchors_l
+
+
+def manual_verify(batch_size, inp, output, in_idx_l, in_anchors_l, shapes_l, out_anchors_l):
+    for i in range(batch_size):
+        out = output.at(i)
+        Y = len(out)
+        X = len(out[0])
+        C = len(out[0][0])
+        for y in range(Y):
+            for x in range(X):
+                col = [None for _ in range(C)]
+                for p in range(len(out_anchors_l[i])):
+                    if out_anchors_l[i][p][0] <= y and y < out_anchors_l[i][p][0] + shapes_l[i][p][0]:
+                        if out_anchors_l[i][p][1] <= x and x < out_anchors_l[i][p][1] + shapes_l[i][p][1]:
+                            col = inp.at(in_idx_l[i][p]) \
+                                [in_anchors_l[i][p][0] + (y - out_anchors_l[i][p][0])] \
+                                [in_anchors_l[i][p][1] + (x - out_anchors_l[i][p][1])]
+                for c in range(C):
+
+                    if col[c] is not None:
+                        if col[c] != out[y][x][c]:
+                            print(i, x, y, col, out[y][x])
+                        assert col[c] == out[y][x][c]
```

Comment:
Doing it pixel by pixel is not a good idea - it's going to be very slow.
Consider something like:
```suggestion
    for i in range(batch_size):
        out = output.at(i)
        assert out.shape == out_size_l[i]
        ref = np.zeros(out.shape)
        for j, idx in enumerate(in_idx_l[i]):
          roi_start = in_anchors_l[i][j]
          roi_end = roi_start + shapes_l[i][j]
          out_start = out_anchors_l[i][j]
          out_end = out_start + shapes_l[i][j]
          ref[out_start[0]:out_end[0], out_start[1]:out_end[1]] = inp[idx][roi_start[0]:roi_end[0], roi_start[1]:roi_end[1]]
          
          assert np.array_equal(out, ref)
```

Reply:
Done